.pre-commit-config.yaml

```diff
@@ -0,0 +1,30 @@
+# File introduces automated checks triggered on git events
```

Comment:
Are these new files? I was thinking we'd also want to rename (i.e. delete) the old `.yml` files.

Reply:
The previous files have been renamed. 
It could be verified [here](https://github.com/Harkirat155/yapf). 
(The PR consists of two commits, one originating from the backlog which has been completely altered by the second since git recognizes renaming as deletion and insertion).

Reply:
Thanks! Yes, I was confused -- we use GitHub Enterprise at work. It's often a bit older version, so things can look different between the two.

Reply:
That's alright. :)
Pleasure. 
🚀